kuberwatcher.py

```diff
@@ -150,12 +152,17 @@ def get_all_pods(namespaces, defaults):
         except:
             print('Could not determine the kind for:', name)
 
-        if kind:
-            kind = kind.lower()
-            pod_group_name = base_name(name, kind)
-            annotations = unflatten(i.metadata.annotations)
-            config = merge_defaults(namespaces.get(namespace, defaults), annotations.get('watcher', {}))
-            kinds[kind][namespace][pod_group_name] = config
+        if not kind:
```

Comment:
Could we just throw the `continue` up in the `except` block above?

Reply:
Could you explain why that would be better? My thought process is that from a logical point of view the `if not kind` is basically "do not continue if `kind` is not set`.

Moving it up results in the same behavior *currently*. However it changes the logic to be specifically "Don't add the pod if this particular except catches it". When we actually want "don't continue if kind isn't set for any reason". 

Moving it up would make it easier for someone to make a change that introduces a new bug accidentally. For example if someone added something to catch a specific exception:

```
try:
    kind = i.metadata.owner_references[0].kind
except NameError:
    print("Variable x is not defined")
except:
    print('Could not determine the kind for:', name)
    continue
```

P.S. Still merging this in as this fixes a bug that is causing some alerts. Please still reply though since we can fix this up in a followup PR as needed.

Reply:
As I was reviewing this code as someone not familiar with it, my mind when to "Why would this be false-y?" and answered "because there was an exception on line 151."

But as I've thought about it more, I've also realized that `i.metadata.owner_references[0].kind` *could* be `None` (or otherwise falsely) even if it didn't throw an exception, and in that case we'd still want to abort this loop iteration early. 

I generally think that `except` handlers should do one of the following:
1) Decide to abort early (by returning from a function, raising a different exception, or `continue`-ing to the next loop interation)
2) Try to "fix" it by having a fallback or default value
3) If it's really not a problem, then it's OK to have a comment + `pass` or print/log a message. 

I think my initial reaction was that this was "1" posing as "3". But in retrospect I think this is fine. 👍🏻